js/coinflex.js

```diff
@@ -0,0 +1,2354 @@
+        //         "status": "COMPLETED",
+        //         "transferredAt": "1651428178967"
+        //      }
+        //
+        // transfer
+        //
+        //     {
+        //         "asset": "USDT",
+        //         "quantity": "5",
+        //         "fromAccount": "38422",
+        //         "toAccount": "38776",
+        //         "transferredAt": "1651704762775"
+        //     }
+        //
+        const currencyId = this.safeString (transfer, 'asset');
+        const timestamp = this.safeString (transfer, 'transferredAt');
+        const fromAccount = this.safeString (transfer, 'fromAccount');
+        const toAccount = this.safeString (transfer, 'toAccount');
+        let status = this.parseTransactionStatus (this.safeString (transfer, 'status'));
+        // if case of 'transfer', as status field is not available, 'transferredAt' prop should mean that transfer was done
+        if (status === undefined && timestamp !== undefined) {
+            status = 'ok';
+        }
+        return {
+            'id': this.safeString (transfer, 'id'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'currency': this.safeCurrencyCode (currencyId, currency),
+            'amount': this.safeNumber (transfer, 'quantity'),
+            'fromAccount': fromAccount,
+            'toAccount': toAccount,
+            'status': status,
+            'info': transfer,
+        };
+    }
+
+    async buildOrderRequest (market, type, side, amount, price = undefined, params = {}) {
+        const clientOrderId = this.safeString (params, 'clientOrderId');
+        const maxCOI = '9223372036854775807';
+        if ((clientOrderId !== undefined) && Precise.stringGt (clientOrderId, maxCOI)) {
+            throw new InvalidOrder (this.id + ' createOrder() param clientOrderId should not exceed ' + maxCOI);
+        }
+        const orderType = this.convertOrderType (type);
+        // creating stop orders using type argument will mess up the unification logic (beacuse of missing market/limit). So, we have to use unified approach for sending stop orders
+        if (orderType === 'STOP') {
+            throw new ArgumentsRequired (this.id + ' createOrder() : to create a stop order, you need to specify the "stopPrice" param');
+        }
+        await this.loadMarkets ();
+        const order = {
+            'marketCode': market['id'],
+            'side': this.convertOrderSide (side),
+            'orderType': this.convertOrderType (type),
+            'quantity': this.amountToPrecision (market['symbol'], amount),
+        };
+        const stopPrice = this.safeNumber (params, 'stopPrice');
+        const isStopOrder = stopPrice !== undefined;
+        if (isStopOrder) {
+            order['stopPrice'] = this.priceToPrecision (market['symbol'], stopPrice);
+            params = this.omit (params, 'stopPrice');
+        }
+        if (price !== undefined && type === 'limit') {
```

Comment:
Currently if type = 'limit' and price = undefined the order will go through and crash on the exchange side. I would suggest refactoring this condition to something like:

```Javascript
if (type === 'limit') {
    if (price === undefined) {
          throw new ArgumentRequired(...)
     }
     // rest of the logic
}

```

Reply:
if I remind it correctly, we should be expecting user to provide limit order without price, and afaik we nowhere handle such occasion. @kroitor can you confirm ?

Reply:
If I was to implement it, i would probably make it short:

```JavaScript
if (type === 'limit') {
    request['price'] = this.priceToPrecision (symbol, price);
    if (price === undefined) {
          throw new ArgumentRequired(...)
     }
     // rest of the logic
}
```

Having this check there isn't critical, but it wouldn't hurt either:

```JavaScript
if (type === 'limit') {
    if (price === undefined) {
          throw new ArgumentRequired(...)
     }
     // rest of the logic
}
```

Reply:
the reason I've not implemented that exception-throwing there, is that we are not doing these things manually in most cases.  maybe we can just defined method in base to facilitate things. Please check [this PR](https://github.com/ccxt/ccxt/pull/13153), as it's better to have that, if we want to check type & price.